src/hotspot/share/runtime/basicLock.hpp

```diff
@@ -28,30 +28,46 @@
 #include "oops/markWord.hpp"
 #include "runtime/atomic.hpp"
 #include "runtime/handles.hpp"
+#include "utilities/globalDefinitions.hpp"
 #include "utilities/sizes.hpp"
 
 class BasicLock {
   friend class VMStructs;
   friend class JVMCIVMStructs;
  private:
+  // * For LM_MONITOR
+  // Unused.
+  // * For LM_LEGACY
   // This is either the actual displaced header from a locked object, or
   // a sentinel zero value indicating a recursive stack-lock.
-  volatile markWord _displaced_header;
+  // * For LM_LIGHTWEIGHT
+  // Used as a cache the ObjectMonitor* used when locking. Must either
+  // be nullptr or the ObjectMonitor* used when locking.
+  volatile uintptr_t _metadata;
```

Comment:
The displaced header/markword terminology was very well known to people, whereas "metadata" is really abstract - people will always need to go and find out what it actually refers to. Could we not define a union here to support the legacy and lightweight modes more explicitly and keep the existing terminology for the setters/getters for the code that uses it?

Reply:
I should have read ahead. I see you do keep the setters/getters.